### PR TITLE
Cross-Compile: macOS X1 (arm64)

### PR DESCRIPTION
**Pull request by ax3l**
* macOS: no arm64 ("X1") native runners yet

Note: this disable Fortran bindings due to a compile error.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
Hm...
If I build a c lib that needs python as a build script dependency but does not build python modules - is this still the right way to express this?
https://github.com/conda-forge/adios-feedstock/blob/8ed8311e0a1d41f08422cc8dfa0890a52d5a0b0b/recipe/conda_build_config.yaml#L10-L15
I updated locally to 3.8 instead of 3.6 but smithy still cannot render this
```
ValueError: variant config in /home/axel/src/conda-forge/adios-feedstock/recipe/conda_build_config.yaml is ambiguous because it
does not fully implement all zipped keys, or specifies a
subspace that is not fully implemented. To be clear:
.. we did not find ['3.8'] from {'mpi': ['nompi', 'mpich', 'openmpi'], 'pin_run_as_build': {'mpich': 'x.x', 'openmpi': 'x.x'}, 'python': ['3.8'], 'ignore_version': ['python']} in python:['3.6.* *_cpython', '3.7.* *_cpython', '3.8.* *_cpython']

```

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
arm64 OSX compile issues with Fortran bindings:
```
$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-gfortran  -g -O2 -c -o core/adiosf_write_mod.o core/adiosf_write_mod.f90
core/adiosf_write_mod.f90:4109:98:
 4109 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_complex, nelems, values, err)
      |                                                                                                  1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (COMPLEX(4)/COMPLEX(8)).
core/adiosf_write_mod.f90:4096:97:

 4096 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double, nelems, values, err)
      |                                                                                                 1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/COMPLEX(8)).
core/adiosf_write_mod.f90:4083:95:

 4083 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_real, nelems, values, err)
      |                                                                                               1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(4)/COMPLEX(8)).
core/adiosf_write_mod.f90:4070:95:

 4070 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_long, nelems, values, err)
      |                                                                                               1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/COMPLEX(8)).
core/adiosf_write_mod.f90:4057:98:

 4057 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_integer, nelems, values, err)
      |                                                                                                  1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)

[...]
```

File: recipe/build.sh [hunk 1]
```diff
@@ -26,12 +26,18 @@ else
    export ADIOS_MPI="--with-mpi=${PREFIX}"
 fi
 
+# Fortran bindings
+export ADIOS_FORTRAN="--enable-fortran"
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
```
Comment (isuruf) [inline recipe/build.sh]:
```suggestion
if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
```

Comment (ax3l) [reply recipe/build.sh]: Ui, forgot to add.

I think I can add `-fallow-argument-mismatch` again to get this compiled...

**Comment by ax3l** [comment]
Is the macOS arm64 OpenMPI hickup due to https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/110 ?

**Comment by isuruf** [comment]
nope. see the changes I did in adios2 to allow cross compilation with openmpi.

**Comment by ax3l** [comment]
Thanks, good point! https://github.com/conda-forge/adios2-feedstock/pull/21

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Pull request merged**